lib/take_elevated.lua

```diff
@@ -41,21 +46,35 @@ local function calculateNewBucketContent(current, tokens_per_ms, bucket_size, cu
     end
 end
 
+local function takeERLQuota(erlQuotaKey, erl_quota_amount, erl_quota_expiration_epoch, current_timestamp_ms)
+    local erlQuotaExists = redis.call('EXISTS', erlQuotaKey)
+    local erlQuota = erl_quota_amount
+    if erlQuotaExists == 1 then
+        erlQuota = tonumber(redis.call('GET', erlQuotaKey))
+    end
+    local newERLQuota = erlQuota - 1
+    if newERLQuota >= 0 then
+        redis.call('SET', erlQuotaKey, newERLQuota, 'PXAT', string.format('%.0f', erl_quota_expiration_epoch))
```

Comment:
@panga 
Just in case, I wanted to double check we're ok to use the PXAT option within the SET command.
I [saw PXAT was added in Redis 6.2](https://raw.githubusercontent.com/redis/redis/6.2/00-RELEASENOTES), and by taking a look at some spaces I see we have version 6.2.6 running, but [checking layer0-base I see the engine_version for aws elasticache is set to 6.x](https://github.com/auth0/layer0-base/blob/721d3577a34cfb99ad1bee981777191b839ab6b4/terraform/v2/modules/aws/elasticache/redis/main.tf#L77)
I can't find the same for Aiven redis.

Reply:
The platform should already support 6.2 and above, in some cases we run 7.x.
It is fine to use this as a requirement, please document in README.